examples/pi-home-router/src/ipv4/classify_dest_subnet.rs

```diff
@@ -0,0 +1,45 @@
+use crate::types::InterfaceAnnotated;
+use cidr::{Cidr, Ipv4Cidr};
+use route_rs_packets::Ipv4Packet;
+use route_rs_runtime::classifier::Classifier;
+use std::collections::HashMap;
+
+/// Classifies an Ipv4Packet by its destination subnet.
+///
+/// This classifier uses a HashMap of CIDRs to subnet names. We require but do not enforce that the
+/// CIDRs are disjoint. Undefined behavior will occur if they are overlapping.
+///
+/// TODO: Actually enforce that CIDRs are disjoint at runtime.
```

Comment:
Actually let's allow overlaps but prefer the longest CIDR to match. We can sort the keys by length once in the constructor for performance.